pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -747,11 +737,6 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) e
 	c.services.handler.Append(func(obj interface{}, event model.Event) error {
 		svc := *obj.(*v1.Service)
 
-		// Do not handle "kube-system" services
-		if svc.Namespace == meta_v1.NamespaceSystem {
-			return nil
-		}
-
 		log.Infof("Handle service %s in namespace %s", svc.Name, svc.Namespace)
 
 		if c.EDSUpdater != nil {
```

Comment:
How about - keep the 'do not handle kube-system' if EDSUpdater == nil ( the old code path ).
Handle it if EDSUpdater is set ?

Reply:
I tried this - it doesn't work. This is *not* the old code path, although it seems like it. Like I mentioned before, EDSUpdater is nil when we create the informer. Maybe not in every possible case, but with Istio 1.0.6 with the default settings it is every time. EDSUpdater is set after this is initialized. Therefor, if we do this, we will handle kube-system, but we will have the optimization turned off, so it will be just as bad as it was before.